Snowflake.Data.Tests/UnitTests/Session/EasyLoggingStarterTest.cs

```diff
@@ -58,9 +70,60 @@ public void BeforeEach()
             t_easyLoggingProvider = new Mock<EasyLoggingConfigProvider>();
             t_easyLoggerManager = new Mock<EasyLoggerManager>();
             t_directoryOperations = new Mock<DirectoryOperations>();
-            t_easyLoggerStarter = new EasyLoggingStarter(t_easyLoggingProvider.Object, t_easyLoggerManager.Object, t_directoryOperations.Object);
+            t_environmentOperations = new Mock<EnvironmentOperations>();
+            t_easyLoggerStarter = new EasyLoggingStarter(t_easyLoggingProvider.Object, t_easyLoggerManager.Object, t_directoryOperations.Object, t_environmentOperations.Object);
         }
-        
+
+        [Test]
+        public void TestThatCreatedDirectoryPermissionsFollowUmask()
+        {
+            // Note: To test with a different value than the default umask, it will have to be set before running this test
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // arrange
+                t_easyLoggingProvider
+                    .Setup(provider => provider.ProvideConfig(ConfigPath))
+                    .Returns(s_configWithInfoLevel);
+                t_directoryOperations
+                    .Setup(provider => provider.Exists(ConfigPath))
+                    .Returns(Directory.Exists(ConfigPath));
+                t_directoryOperations
+                    .Setup(provider => provider.CreateDirectory(s_expectedLogPath))
+                    .Returns(Directory.CreateDirectory(s_expectedLogPath));
+
+                // act
+                t_easyLoggerStarter.Init(ConfigPath);
+                var umask = EasyLoggerUtil.AllPermissions - int.Parse(EasyLoggerUtil.CallBash("umask"));
```

Comment:
umask is only int.Parse(EasyLoggerUtil.CallBash("umask")) e. g. 022.
This whole expression should be called rather expectedDirPermissions e. g. 755.
Also I would move expectedDirPermissions calculation to the arrange block.

Reply:
Renamed and moved

Reply:
Fixed